components/ConnectButton.js

```diff
@@ -1,32 +1,21 @@
 import Button from './ui/Button'
 import useWeb3 from './hooks/useWeb3'
 import { Body1 } from './ui/Typography'
-import { NetworkId } from '../util/constants'
+import { Network } from '../util/constants'
 
 const ConnectButton = () => {
-  const { connect, accounts, error, provider, networkId } = useWeb3()
+  const { connect, address, networkError, web3 } = useWeb3()
 
-  if (error) {
-    if (networkId !== NetworkId) {
-      return (
-        <Button
-          onClick={async () => {
-            await provider.request({
-              method: 'wallet_switchEthereumChain',
-              params: [{ chainId: NetworkId }],
-            })
-          }}
-        >
-          {error}
-        </Button>
-      )
-    }
-
-    return <Body1>{error}</Body1>
+  if (networkError) {
+    web3.request({
+      method: 'wallet_switchEthereumChain',
+      params: [{ chainId: Network.hexId }],
+    })
+    return <Button>{networkError}</Button>
```

Comment:
The earlier version had the chain switch as the `onClick`; if that’s no longer needed and we’re just going to invoke it, maybe the error should go to a different element, `Body1` but red perhaps?

Reply:
@katyjeremko had asked for this, any thoughts here?